src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -295,53 +295,52 @@ private Assembly Resolve(AssemblyLoadContext loadContext, AssemblyName assemblyN
         // If there are multiple version of the assembly, load the latest.
         private bool TryFindInGAC(AssemblyName assemblyName, out string assemblyFilePath)
         {
-            bool assemblyFound = false;
             assemblyFilePath = null;
+            if (Internal.InternalTestHooks.DisableGACLoading) { return false; }
+
+            bool assemblyFound = false;
             char dirSeparator = IO.Path.DirectorySeparatorChar;
 
-            if (PowerShellAssemblyLoadContextTestHooks.AllowGACLoading)
+            if(String.IsNullOrEmpty(_winDir))
```

Comment:
looks like this file is inconsistent on whitespace after the `if`, at least for your changes, you should probably always have the whitespace